snap/snapenv/snapenv.go

```diff
@@ -85,6 +85,7 @@ func basicEnv(info *snap.Info) osutil.Environment {
 		"SNAP":               filepath.Join(dirs.CoreSnapMountDir, info.SnapName(), info.Revision.String()),
 		"SNAP_COMMON":        snap.CommonDataDir(info.SnapName()),
 		"SNAP_DATA":          snap.DataDir(info.SnapName(), info.Revision),
+		"SNAP_SAVE_DATA":     snap.CommonSaveDir(info.InstanceName()),
```

Comment:
this should be set only if the directory exists

Reply:
Done! I add this env only if the path exists now.